src/ring_theory/localization.lean

```diff
@@ -2219,6 +2228,20 @@ lemma is_unit_map_of_injective (hg : function.injective g)
 is_unit.mk0 (g y) $ show g.to_monoid_with_zero_hom y ≠ 0,
   from map_ne_zero_of_mem_non_zero_divisors g hg y.2
 
+lemma eq_zero_of_mk'_eq_zero [algebra R K] [is_fraction_ring R K] {x : R} {y : non_zero_divisors R}
+ (hxy : mk' K x y = 0) : x = 0 :=
+begin
+  simp_rw [mk'_eq_zero_iff, mul_right_coe_non_zero_divisors_eq_zero_iff] at hxy,
+  exact (exists_const _).mp hxy
+end
+
+lemma eq_of_mk'_eq_one {x : A} {y : non_zero_divisors A} (hxy : mk' K x y = 1) : x = y :=
+begin
+  rw [is_fraction_ring.mk'_eq_div, div_eq_one_iff_eq] at hxy,
```

Comment:
The `oleans` are still being built so I can't look at this in detail but the code suggests one of these rewrites introduces an additional goal, which is a bit undesirable. Is there an easy refactoring to avoid this happening?

Incidentally, this will not be relevant if you refactor as above but there is an `exacts` tactic for closing multiple goals like this. In this case it would be:
```lean
  exacts [is_fraction_ring.injective A K hxy, is_fraction_ring.to_map_ne_zero_of_mem_non_zero_divisors y.property],
```
though as I say, hopefully we can avoid introducing the second goal.

Reply:
I proved the second goal before the `rw` (I don't think I can inline it because Lean cannot figure out the right type).